File: client/real-estate-crm-react/src/types/deal.ts
export const DealStatus = {
  Pending: "Pending",
  Reserved: "Reserved",
  Contracted: "Contracted",
  Cancelled: "Cancelled",
} as const;
export type DealStatus = (typeof DealStatus)[keyof typeof DealStatus];

export interface Deal {
  id: string;
  leadId: string;
  unitId: string;
  salesAgentId: string;
  dealValue: number;
  status: DealStatus;
  reservationDate: string | null;
  contractDate: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateDealRequest {
  leadId: string;
  unitId: string;
  salesAgentId?: string | null;
  dealValue: number;
  notes?: string | null;
}

export interface UpdateDealRequest {
  dealValue: number;
  notes?: string | null;
}

export interface DealListQuery {
  page?: number;
  pageSize?: number;
  status?: DealStatus;
  leadId?: string;
  unitId?: string;
  salesAgentId?: string;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}

File: mobile/lib/features/deals/domain/deal.dart
class Deal {
  const Deal({
    required this.id,
    required this.leadId,
    required this.unitId,
    required this.salesAgentId,
    required this.dealValue,
    required this.status,
    required this.reservationDate,
    required this.contractDate,
    required this.notes,
  });

  factory Deal.fromJson(Map<String, dynamic> json) => Deal(
    id: json['id'] as String,
    leadId: json['leadId'] as String,
    unitId: json['unitId'] as String,
    salesAgentId: json['salesAgentId'] as String,
    dealValue: (json['dealValue'] as num).toDouble(),
    status: json['status'] as String,
    reservationDate: json['reservationDate'] as String?,
    contractDate: json['contractDate'] as String?,
    notes: json['notes'] as String?,
  );

  final String id;
  final String leadId;
  final String unitId;
  final String salesAgentId;
  final double dealValue;
  final String status;
  final String? reservationDate;
  final String? contractDate;
  final String? notes;
}
